fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -914,8 +918,20 @@ protected void unprotectedCommitTransaction(TransactionState transactionState, S
         if (transactionState.getTransactionStatus() != TransactionStatus.PREPARE) {
             return;
         }
+        // since we send publish order by commit timestamp
+        // so that we need handle timetamp fallback
+        // & same timestamp cause by granularity
+        // The probability of timestamp fallback after FE failover is small
+        // and it is not considered at present
+        long commitTs = System.currentTimeMillis();
+        if (commitTs <= lastCommitTs) {
+            commitTs = lastCommitTs + ++commitTsInc;
```

Comment:
How about the FE restart?
Is there necessary to persist the commitTsInc and the commitTs?

Reply:
I have already commented
since the probability is low and not a fatal error, not necessary to persist